### ISSUE TITLE
[10.0.0-beta.0] onScroll doesn't work

### ISSUE THREAD

**Issue by kozakl** [developit/preact#1551]
Hello.
I discovered, that onScroll event doesn't work in preact (10.0.0-beta.0), event even doesn't call handler.
Thanks.

**Comment by JoviDeCroock** [comment]
Seems to work here, could you reproduce an example for your case: https://codesandbox.io/s/xv9xkv6494

**Comment by kozakl** [comment]
Thanks man. 
Your example works, but see mine example:

preact:
https://codesandbox.io/embed/kw8xm62x9r

react:
https://codesandbox.io/embed/m79797jn0j

there is something wrong with preact, when you scroll horizontal, in preact version you don't see that onScroll works, but in react version everything works like a harm.

thanks

**Comment by cristianbote** [comment]
Hey @kozakl,

**TL;DR;**
So, in your examples, all you need to do is define the `onScroll` listener on the `div` that will _actually_ trigger the scroll event.
Instead of this:
```jsx
<div onScroll={() => console.log("scroll")}>
      <div style={{ overflowX: "scroll", whiteSpace: "nowrap" }}>
```

Change it to this:
```jsx
<div>
      <div onScroll={() => console.log("scroll")} style={{ overflowX: "scroll", whiteSpace: "nowrap" }}>
```

The longer version, actually is about the event `capture` phase. By _capturing_ the event at a parent level you rely on the event to bubble.
By default, the events are added with `useCapture: false`. That can be specified with the `Capture` suffix. So, in your example, you could define `onScrollCapture` at the parent level to capture the scroll event at your top parent level.

So, why does `React` work like a charm? React has an internal events map, that needs to be captured or not. In your case `scroll` is part of that mapping. https://github.com/facebook/react/blob/66f280c87b05885ee55320a5e107a534a50f9375/packages/react-dom/src/events/DOMTopLevelEventTypes.js#L125
Then, it gets _trapped_ https://github.com/facebook/react/blob/66f280c87b05885ee55320a5e107a534a50f9375/packages/react-dom/src/events/ReactBrowserEventEmitter.js#L140.
Bottom line, React does the capturing for you. With Preact, you need to be specific about the events that you want to capture or not.

Hopefully this clears the air about, why and how it works.

**Comment by kozakl** [comment]
@cristianbote Thanks dude, now everything is clear!
